src/matricize.jl

```diff
@@ -0,0 +1,123 @@
+using TensorProducts: ⊗
+
+# =====================================  FusionStyle  ======================================
+abstract type FusionStyle end
+
+struct ReshapeFusion <: FusionStyle end
+
+FusionStyle(a::AbstractArray) = FusionStyle(a, axes(a))
+function FusionStyle(a::AbstractArray, t::Tuple{Vararg{AbstractUnitRange}})
+  return FusionStyle(a, combine_fusion_styles(FusionStyle.(t)...))
+end
+
+# Defaults to ReshapeFusion, a simple reshape
+FusionStyle(::AbstractArray{<:Any,0}) = ReshapeFusion()   # TBD better solution?
+FusionStyle(::AbstractUnitRange) = ReshapeFusion()
+FusionStyle(::AbstractArray, ::ReshapeFusion) = ReshapeFusion()
+
+combine_fusion_styles(::Style, ::Style) where {Style<:FusionStyle} = Style()
+combine_fusion_styles(::FusionStyle, ::FusionStyle) = ReshapeFusion()
+combine_fusion_styles(styles::FusionStyle...) = foldl(combine_fusion_styles, styles)
+
+# =======================================  misc  ========================================
+function fuseaxes(
+  axes::Tuple{Vararg{AbstractUnitRange}}, blockedperm::AbstractBlockPermutation
+)
+  axesblocks = blockpermute(axes, blockedperm)
+  return map(block -> ⊗(block...), axesblocks)
+end
+
+Base.permutedims(a::AbstractArray, bp::AbstractBlockPermutation) = permutedims(a, Tuple(bp))
+Base.permutedims(a::StridedArray, bp::AbstractBlockPermutation) = permutedims(a, Tuple(bp))
```

Comment:
Isn't this covered by the version above? Is it to solve a method ambiguity? If so please add a comment about that.